### PR TITLE
Adding code for Pods to be cleaned up by TTL Controller

### PR DESCRIPTION
**Pull request by techwhizbang**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test

/kind feature

> /kind flake

**What this PR does / why we need it**:

This PR cleans up Pods for the TTL Controller
For: kubernetes/enhancements#592

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

@kubernetes/sig-apps-pr-reviews

This is a continuation of https://github.com/kubernetes/kubernetes/pull/82082 whereby the comments and suggested improvement of the previous code reviewers have been addressed.

**Does this PR introduce a user-facing change?**:

NONE

```release-note
Added code for pods to be cleaned up by TTLAfterFinished Controller
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
[KEP]: https://github.com/kubernetes/enhancements/blob/master/keps/sig-apps/0026-ttl-after-finish.md
```

### CODE REVIEW

**Comment by lavalamp** [comment]
Just FYI I'm pretty busy and probably won't be able to review this for a
while. You might want to find others to help with that.

**Comment by techwhizbang** [comment]
@lavalamp thanks for the heads up - any suggestions on who might be able to assist instead?

**Comment by techwhizbang** [comment]
@gianarb @liggitt wondering if you could help me move this forward as you were assisting with the code review on the now outdated PR #82082? this is a superset of that PR and containing the suggested improvements and also rebasing required.

**Comment by wojtek-t** [comment]
/cc

@techwhizbang - TBD, with upcoming codefreeze, I doubt we will be able to have that in for 1.19 (I won't have time to look for sure by then).

**Comment by fedebongio** [comment]
/assign @seans3 
Sean, do you think someone in your SIG can take a look?
/unassign @lavalamp 
/cc @lavalamp

File: pkg/controller/ttlafterfinished/ttlafterfinished_controller.go [hunk 1]
```diff
@@ -82,135 +93,250 @@ func New(jobInformer batchinformers.JobInformer, client clientset.Interface) *Co
...
-// finishes and its TTL after finished has expired. If the Job hasn't finished or
-// its TTL hasn't expired, it will be added to the queue after the TTL is expected
-// to expire.
+// processResource will check the Resource's [job / pod] state and TTL and
+// delete the resource when it finishes and its TTL after finished has expired.
+// If the resource hasn't finished or its TTL hasn't expired, it will be added
+// to the queue after the TTL is expected to expire.
 // This function is not meant to be invoked concurrently with the same key.
-func (tc *Controller) processJob(key string) error {
+func (tc *Controller) processResource(key string) error {
```
Comment (janetkuo) [inline pkg/controller/ttlafterfinished/ttlafterfinished_controller.go:273]: We could process the wrong resource if a Job has the same key (namespace + name) as a Pod. How do we resolve that? The key needs to differentiate that.

**Pull request closed without merging**